### DISCUSSION TITLE
Dialog as nested page lead to scroll to top on parent page

### DISCUSSION THREAD

**Discussion by awacode21** [nuxt/nuxt#25218] [Questions]
Hi Guys,

i need your help!
I have a nested page to show a dialog and also be able to open up that dialog on page entry by route and to have it as part of my back/forth history. So i chose that nested page would be the way to go. And in general it works fine. On route change to the `/details` nested route it opens and changing back to root route `/` it closes.

But i have one very annoying issue. Every time i call my nested route to open the overlay, the page gets scrolled to top. But in my case with the dialog i don't want the user to loose its context. So in case of my `/details` nested page i want to prevent that scroll to top. But for other "non nested" routes i would like to keep it.

I already tried to use
`definePageMeta({ scrolToTop: false });`

on the nested pages. But without success.
I only want to disable this for dialog cases.. for other route changes it should stay with the default behaviour.
Any idea how to handle this?

**Example:**
https://stackblitz.com/edit/nuxt-starter-sg3vvg

**Comment by awacode21** [comment]
Well i can answer my question by myself. I had a typo in definePageMeta. Correct will be:

`definePageMeta({ scrollToTop: false });`

on nested page. Then it works as expected.

**Reply by awacode21** [reply]
I need to unmark this answer as after using it a while i figured out it does not work as expected. 
It works on first call of nested route view. But when i go back to root page, scroll up and down to somewhere and again open nested dialog page it gets scrolled to scroll position the root page had the last time i went to nested page. So it scrolls to second last scroll position and ignores that i set `definePageMeta({ scrollToTop: false });` to not scroll at all.

I added some more "open dialog" buttons within my example at different places on the page to show the behaviour. 

Is this a Nuxt bug or wrong usage? Shouldn't it never scroll for nested page when i define `definePageMeta({ scrollToTop: false });`? and when it ignores this setting  i guess it should not use second last scroll position but the last one.

**Reply by awacode21** [reply]
i created a bug ticket: [https://github.com/nuxt/nuxt/issues/25241](https://github.com/nuxt/nuxt/issues/25241) with an updated stackblitz example: [https://stackblitz.com/edit/nuxt-starter-5xeqgj](https://stackblitz.com/edit/nuxt-starter-5xeqgj)

**Comment by gbissolli** [comment]
Hey, I am having the same issue... Have you got any updates on this?  Thanks in advance.  @awacode21
